src/mscorlib/src/System/AppDomainSetup.cs

```diff
@@ -213,230 +208,29 @@ public String ApplicationBase
-
-                    if ((appBase == null) || (appBase.Length == 0))
-                        throw new MemberAccessException(SR.AppDomain_AppBaseNotSet);
-
-                    StringBuilder result = StringBuilderCache.Acquire();
-
-                    bool slash = false;
-                    if ((path[0] == '/') || (path[0] == '\\'))
-                    {
-                        string pathRoot = AppDomain.NormalizePath(appBase, fullCheck: false);
-                        pathRoot = pathRoot.Substring(0, IO.PathInternal.GetRootLength(pathRoot));
-
-                        if (pathRoot.Length == 0)
-                        { // URL
-                            int index = appBase.IndexOf(":/", StringComparison.Ordinal);
-                            if (index == -1)
-                                index = appBase.IndexOf(":\\", StringComparison.Ordinal);
-
-                            // Get past last slashes of "url:http://"
-                            int urlLen = appBase.Length;
-                            for (index += 1;
-                                 (index < urlLen) && ((appBase[index] == '/') || (appBase[index] == '\\'));
-                                 index++) ;
-
-                            // Now find the next slash to get domain name
-                            for (; (index < urlLen) && (appBase[index] != '/') && (appBase[index] != '\\');
-                                index++) ;
-
-                            pathRoot = appBase.Substring(0, index);
-                        }
-
-                        result.Append(pathRoot);
-                        slash = true;
-                    }
-                    else
-                        result.Append(appBase);
-
-                    // Make sure there's a slash separator (and only one)
-                    int aLen = result.Length - 1;
-                    if ((result[aLen] != '/') &&
-                        (result[aLen] != '\\'))
-                    {
-                        if (!slash)
-                        {
-#if !PLATFORM_UNIX
-                            if (appBase.IndexOf(":/", StringComparison.Ordinal) == -1)
-                                result.Append('\\');
-                            else
-#endif // !PLATFORM_UNIX
-                                result.Append('/');
-                        }
-                    }
-                    else if (slash)
-                        result.Remove(aLen, 1);
-
-                    result.Append(path);
-                    path = StringBuilderCache.GetStringAndRelease(result);
-                }
-                else
-                    path = AppDomain.NormalizePath(path, fullCheck: true);
+                Value[(int)LoaderInformation.ApplicationBaseValue] = (value == null || value.Length == 0)?null:Path.GetFullPath(value);
             }
-
-            return path;
         }
-
+        
         public String ConfigurationFile
         {
```

Comment:
Bonus points for deleting all other properties in AppDomainSetup that are not used anywhere :-)